fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -2642,16 +2657,24 @@ public void createDb(CreateDbStmt stmt) throws DdlException {
                     ErrorReport.reportDdlException(ErrorCode.ERR_DB_CREATE_EXISTS, fullDbName);
                 }
             } else {
-                id = getNextId();
-                Database db = new Database(id, fullDbName);
-                db.setClusterName(clusterName);
                 unprotectCreateDb(db);
                 editLog.logCreateDb(db);
             }
         } finally {
             unlock();
         }
         LOG.info("createDb dbName = " + fullDbName + ", id = " + id);
+
+        // create tables in iceberg database
+        if (db.getDbProperties().getIcebergProperty().isExist()) {
+            IcebergProperty icebergProperty = db.getDbProperties().getIcebergProperty();
+            IcebergCatalog icebergCatalog = IcebergCatalogMgr.getCatalog(icebergProperty);
+            List<TableIdentifier> icebergTables = icebergCatalog.listTables(icebergProperty.getDatabase());
```

Comment:
What if the `listTables` throw exception?

Reply:
not resolved?

Reply:
Added a new `DorisIcebergException` which is a `RuntimeException` in `IcebergCatalog.listTable`.